pkg/datastore/mysql/iterator.go

```diff
@@ -35,10 +40,89 @@ func (it *Iterator) Next(dst interface{}) error {
 	if err != nil {
 		return err
 	}
+
+	// Update last iterated item as last read row.
+	it.last = &rowDataConverter{val: val}
+
 	return decodeJSONValue(val, dst)
 }
 
-// Cursor implementation for MySQL Iterator
+// Cursor builds a base 64 string (encode from string in map[string]interface{} format).
+// The cursor contains only values attached with the fields used
+// as ordering fields.
 func (it *Iterator) Cursor() (string, error) {
-	return "", datastore.ErrUnimplemented
+	if it.last == nil {
+		return "", datastore.ErrInvalidCursor
+	}
+
+	lastObjData := it.last.Data()
+
+	cursor := make(map[string]interface{}, len(it.orders))
+	for _, o := range it.orders {
+		val, ok := lastObjData[o.Field]
+		if !ok {
+			return "", datastore.ErrInvalidCursor
+		}
+		// TODO: Support build cursor from nested Ordering field.
+		cursor[o.Field] = val
+	}
+
+	b, _ := json.Marshal(cursor)
+	return base64.StdEncoding.EncodeToString(b), nil
+}
+
+type DataConverter interface {
```

Comment:
Can this be a private interface?

Reply:
Sure, as discussed at this comment https://github.com/pipe-cd/pipe/pull/1873#discussion_r615545623
Currently, we have 2 options for this
- make this `DataConverter` as datastore package interface and share for both MySQL and Firestore
- make both Firestore and MySQL DataConverter as a private interface 
To be honest, I think shared one could be better, I will rename it as `LastIterator` interface or something 🤔 
How do you think about that 👀

Reply:
That interface is just an additional thing for the cursor, for testing so I think we should define them as private in each package instead of defining a new shared one in the datastore package.
Actually, the caller doesn't care about that Interface. That keeps our datastore package simple as it is.

Reply:
I get your point 👍 Addressed on https://github.com/pipe-cd/pipe/pull/1873/commits/dc749f547585cf23b6341ae21681efd317a30ba1 🙏 

Will address the same thing with Firestore iterator cursor by a separated PR later 🙏